### ISSUE TITLE
Incorrect interaction between `rtl` languages and table columns

### ISSUE THREAD

**Issue by A-Walrus** [typst/typst#3373]
### Description

When using an `rtl` language /  setting the text dir to `rtl` directly,  table columns go right to left, as they should. But their numbering is still left to right, as can be observed when using a fill function.

```typ
#set text(dir:rtl)

#table( 
  columns: (1fr,1fr,1fr),
  fill: (col,row) => {
    if col == 0 {
      red
    }
    else {
      white
    }
  },
  [1],
  [2],
  [3],
  [1],
  [2],
  [3],
  [1],
  [2],
  [3],
  [1],
  [2],
  [3],
  [1],
  [2],
  [3],
  [1],
  [2],
  [3],
)
```

![image](https://github.com/typst/typst/assets/58790821/bcf1aa3f-e482-4987-97f2-4c8cd1ef716a)

In this example I would expect the first column (with 1s) to be red, not the last column.

### Reproduction URL

https://typst.app/project/pA69dJ-jBVCi95eIyQf31N

### Operating system

Web app, Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by PgBiel** [comment]
Likely related to #1386 , which was fixed with #3037 . Table RTL in particular was also reworked in #3239 .

So, this should be fixed in the next update, as seen below. :)

![the colors are correct now](https://github.com/typst/typst/assets/9021226/a484059e-f003-4d6b-8441-5c8fe5e92f88)

**Comment by A-Walrus** [comment]
Great, Thanks!
